contributors/design-proposals/storage/csi-snapshot.md

```diff
@@ -0,0 +1,316 @@
+Kubernetes CSI Snapshot Proposal
+================================
+
+**Authors:** [Jing Xu](https://github.com/jingxu97), [Xing Yang](https://github.com/xing-yang), [Tomas Smetana](https://github.com/tsmetana), [Huamin Chen ](https://github.com/rootfs)
+
+## Background
```

Comment:
There should be some relation to the previous snapshot implementation. Will there be a supported path how to move from the old implementation to the new one?

Reply:
The previous snapshot implementation is not for CSI.  Is there a way to move PV/PVC from non-CSI drivers to CSI drivers?  If there's a way, then maybe we can do similar move for snapshots.  Maybe some kind of export/import?

Reply:
I'll take a look of the migration of in-tree plugin to CSI spec.  Maybe we can do something similar here.

Reply:
I am not advocating for full backward compatibility, I think a paragraph or two about relation of these two implementations and perhaps high-level list of (manual) steps necessary to move from the old to the new one would be enough.

Reply:
Sure.

Reply:
@jsafrane I added a section "Transition to the New Snapshot Support" to address your comments.  Please take a look.  Thanks.